x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplication.java

```diff
@@ -239,17 +237,14 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) return false;
         SearchApplication app = (SearchApplication) o;
         return name.equals(app.name)
-            && Arrays.equals(indices, app.indices)
```

Comment:
I don't think we should take indices out of `equals` and `hashCode` here since we are storing them as part of the object? This also could potentially impact backwards compatibility?

Reply:
this change was made to mitigate the serialization BWC failures.

Reply:
I don't think that's what we want to do here. I think in the BWC test we need to check the version and construct the appropriate object based on the version.